app/uk/gov/hmrc/gform/sharedmodel/ServiceCallResponse.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.sharedmodel
+
+import play.api.libs.json._
+
+sealed trait ServiceCallResponse[+A] extends Product with Serializable
```

Comment:
why do we even need this? seems like this could be avoided? if its all fine just return 200? and for others appropriate code?

Reply:
This code exists to lift http protocol intricacies into type level. 

In a long run I'd like to see migration of all our calls to third party apis (ie. not platform calls or destinations related) to this data type.